src/multibase/multibase.test.ts

```diff
@@ -0,0 +1,128 @@
+    ["base32", "fooba", "bmzxw6ytb"],
+    ["base32", "foobar", "bmzxw6ytboi"],
+    ["base32pad", "yes mani !", "cpfsxgidnmfxgsibb"],
+    ["base32pad", "f", "cmy======"],
+    ["base32pad", "fo", "cmzxq===="],
+    ["base32pad", "foo", "cmzxw6==="],
+    ["base32pad", "foob", "cmzxw6yq="],
+    ["base32pad", "fooba", "cmzxw6ytb"],
+    ["base32pad", "foobar", "cmzxw6ytboi======"],
+    ["base32hex", "yes mani !", "vf5in683dc5n6i811"],
+    ["base32hex", "f", "vco"],
+    ["base32hex", "fo", "vcpng"],
+    ["base32hex", "foo", "vcpnmu"],
+    ["base32hex", "foob", "vcpnmuog"],
+    ["base32hex", "fooba", "vcpnmuoj1"],
+    ["base32hex", "foobar", "vcpnmuoj1e8"],
+    ["base32hexpad", "yes mani !", "tf5in683dc5n6i811"],
+    ["base32hexpad", "f", "tco======"],
+    ["base32hexpad", "fo", "tcpng===="],
+    ["base32hexpad", "foo", "tcpnmu==="],
+    ["base32hexpad", "foob", "tcpnmuog="],
+    ["base32hexpad", "fooba", "tcpnmuoj1"],
+    ["base32hexpad", "foobar", "tcpnmuoj1e8======"],
+    ["base32z", "yes mani !", "hxf1zgedpcfzg1ebb"],
+    ["base58flickr", "yes mani !", "Z7Pznk19XTTzBtx"],
+    ["base58btc", "yes mani !", "z7paNL19xttacUY"],
+    ["base64", "÷ïÿ", "mw7fDr8O/"],
+    ["base64", "f", "mZg"],
+    ["base64", "fo", "mZm8"],
+    ["base64", "foo", "mZm9v"],
+    ["base64", "foob", "mZm9vYg"],
+    ["base64", "fooba", "mZm9vYmE"],
+    ["base64", "foobar", "mZm9vYmFy"],
+    ["base64", "÷ïÿ🥰÷ïÿ😎🥶🤯", "mw7fDr8O/8J+lsMO3w6/Dv/CfmI7wn6W28J+krw"],
+    ["base64pad", "f", "MZg=="],
+    ["base64pad", "fo", "MZm8="],
+    ["base64pad", "foo", "MZm9v"],
+    ["base64pad", "foob", "MZm9vYg=="],
+    ["base64pad", "fooba", "MZm9vYmE="],
+    ["base64pad", "foobar", "MZm9vYmFy"],
+    ["base64url", "÷ïÿ", "uw7fDr8O_"],
+    ["base64url", "÷ïÿ🥰÷ïÿ😎🥶🤯", "uw7fDr8O_8J-lsMO3w6_Dv_CfmI7wn6W28J-krw"],
+    ["base64urlpad", "f", "UZg=="],
+    ["base64urlpad", "fo", "UZm8="],
+    ["base64urlpad", "foo", "UZm9v"],
+    ["base64urlpad", "foob", "UZm9vYg=="],
+    ["base64urlpad", "fooba", "UZm9vYmE="],
+    ["base64urlpad", "foobar", "UZm9vYmFy"],
+    [
+      "base64urlpad",
+      "÷ïÿ🥰÷ïÿ😎🥶🤯",
+      "Uw7fDr8O_8J-lsMO3w6_Dv_CfmI7wn6W28J-krw==",
+    ],
+  ];
+
+  for (const [name, input, expectedOutput] of testCases) {
+    const encoded = encode(name, encodeText(input));
+    assertEquals(
+      decodeText(encoded),
+      expectedOutput,
+      `Encoding ${name} failed`,
+    );
+
+    const decoded = decode(expectedOutput);
+    assertEquals(decoded, encodeText(input), `Decoding ${name} failed`);
+
+    const decodedFromBuffer = decode(encodeText(expectedOutput));
+    assertEquals(
+      decodedFromBuffer,
+      encodeText(input),
+      `Decoding buffer of ${name} failed`,
+    );
+  }
+
+  test("should allow base32pad full alphabet", () => {
+    const encodedStr = "ctimaq4ygg2iegci7";
+    const decoded = decode(encodedStr);
+    const encoded = encode("c", decoded);
+    assertEquals(decode(encoded), decoded);
+  });
```

Comment:
To break down tests into smaller parts, you should use `t.step()` method:

```suggestion
  await t.step("should allow base32pad full alphabet", () => {
    const encodedStr = "ctimaq4ygg2iegci7";
    const decoded = decode(encodedStr);
    const encoded = encode("c", decoded);
    assertEquals(decode(encoded), decoded);
  });
```

To get a `t` object, you should change the first line of this test:

```ts
test("multibase.encode and decode", async (t) => {
```

See also <https://docs.deno.com/runtime/fundamentals/testing/#test-steps>.

Reply:
Thank you for the patience and the link 🙏

Reply:
Is this too much granularity do you think?

```ts
./runtime/multibase/multibase.test.ts => multibase.encode and decode ... Encoding/Decoding base32 with yes mani ! ... ok (0ms)
```

Reply:
I think it's fine?